:rotating_light: :rotating_light: :snake:  Source Google Ads: fix custom queries

tolik0 opened the pull request:
<!--
Thanks for your contribution! 
Before you submit the pull request, 
I'd like to kindly remind you to take a moment and read through our guidelines
to ensure that your contribution aligns with the type of contributions our project accepts.
All the information you need can be found here:
   https://docs.airbyte.com/contributing-to-airbyte/

We truly appreciate your interest in contributing to Airbyte,
and we're excited to see what you have to offer! 

If you have any questions or need any assistance, feel free to reach out in #contributions Slack channel.
-->

## What
Resolved issues with schema creation for custom queries.
[#30253](https://github.com/airbytehq/airbyte/issues/30253)

## How
- Corrected array type handling in custom queries.
- Replaced _protobuf_message_ in the Message type with a string representation, as it is unnecessary.

## 🚨 User Impact 🚨
This update introduces breaking changes. Users must refresh their schemas due to custom query schemas creation changes. For example, for field `ad_group_ad.ad.final_urls` from custom query type change from `{"type": "string"} `to `{"type": ["null", "array"],  "items": {"type": "string"}}`. [Breaking changes doc.](https://docs.google.com/document/d/1Lp8fCo7Fb7brnUD_yoeBcYWNveivxSp_FfIuvwB6Fm8/edit)

airbyte-oss-build-runner commented:
## <img src="https://raw.githubusercontent.com/airbytehq/airbyte/f8bea7f05a6b9d6d8475b181412ecd198080222b/airbyte-integrations/connectors/source-google-ads/icon.svg" width="40" height="40"> source-google-ads test report (commit [`f8bea7f05a`](https://github.com/airbytehq/airbyte/pull/30705/commits/f8bea7f05a6b9d6d8475b181412ecd198080222b)) - ✅

⏲️  Total pipeline duration: 07mn19s 

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Connector package install                                                | ✅        |
| Build source-google-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ✅        |
| Code format checks                                                       | ✅        |
| Validate airbyte-integrations/connectors/source-google-ads/metadata.yaml | ✅        |
| Connector version semver check                                           | ✅        |
| Connector version increment check                                        | ✅        |
| QA checks                                                                | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/tolik0_source-google-ads_custom-query-fix/1695424044/f8bea7f05a6b9d6d8475b181412ecd198080222b/source-google-ads/0.11.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:f8bea7f05a6b9d6d8475b181412ecd198080222b)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-ads test
```

davydov-d reviewed (changes requested):
Can you please add more context here? Why are you removing `protobuf_message` from the code? Why is this a breaking change? If it is, we should bump major version and follow a breaking change release process

tolik0 commented:
> Can you please add more context here? Why are you removing `protobuf_message` from the code? Why is this a breaking change? If it is, we should bump major version and follow a breaking change release process

@davydov-d I updated description and added migration guide.

airbyte-oss-build-runner commented:
## <img src="https://raw.githubusercontent.com/airbytehq/airbyte/e4eeaa2c6cba19deee5a676fd1cff20cd4b9d5b0/airbyte-integrations/connectors/source-google-ads/icon.svg" width="40" height="40"> source-google-ads test report (commit [`e4eeaa2c6c`](https://github.com/airbytehq/airbyte/pull/30705/commits/e4eeaa2c6cba19deee5a676fd1cff20cd4b9d5b0)) - ❌

⏲️  Total pipeline duration: 01mn22s 

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Connector package install                                                | ✅        |
| Build source-google-ads docker image for platform linux/x86_64           | ❌        |
| Code format checks                                                       | ✅        |
| Validate airbyte-integrations/connectors/source-google-ads/metadata.yaml | ✅        |
| Connector version semver check                                           | ✅        |
| Connector version increment check                                        | ✅        |
| QA checks                                                                | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/tolik0_source-google-ads_custom-query-fix/1695675216/e4eeaa2c6cba19deee5a676fd1cff20cd4b9d5b0/source-google-ads/0.11.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:e4eeaa2c6cba19deee5a676fd1cff20cd4b9d5b0)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-ads test
```

airbyte-oss-build-runner commented:
## <img src="https://raw.githubusercontent.com/airbytehq/airbyte/67f87f15aa7abac5379b528874c2fd4bfc8b2281/airbyte-integrations/connectors/source-google-ads/icon.svg" width="40" height="40"> source-google-ads test report (commit [`67f87f15aa`](https://github.com/airbytehq/airbyte/pull/30705/commits/67f87f15aa7abac5379b528874c2fd4bfc8b2281)) - ❌

⏲️  Total pipeline duration: 06mn36s 

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Connector package install                                                | ✅        |
| Build source-google-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ❌        |
| Code format checks                                                       | ✅        |
| Validate airbyte-integrations/connectors/source-google-ads/metadata.yaml | ✅        |
| Connector version semver check                                           | ✅        |
| Connector version increment check                                        | ✅        |
| QA checks                                                                | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/tolik0_source-google-ads_custom-query-fix/1695676055/67f87f15aa7abac5379b528874c2fd4bfc8b2281/source-google-ads/1.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:67f87f15aa7abac5379b528874c2fd4bfc8b2281)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-ads test
```

airbyte-oss-build-runner commented:
## <img src="https://raw.githubusercontent.com/airbytehq/airbyte/d5e7f6b60efc7d31acf2556bff730735509073d1/airbyte-integrations/connectors/source-google-ads/icon.svg" width="40" height="40"> source-google-ads test report (commit [`d5e7f6b60e`](https://github.com/airbytehq/airbyte/pull/30705/commits/d5e7f6b60efc7d31acf2556bff730735509073d1)) - ✅

⏲️  Total pipeline duration: 08mn01s 

| Step                                                           | Result   |
|:---------------------------------------------------------------|:---------|
| Connector package install                                      | ✅        |
| Build source-google-ads docker image for platform linux/x86_64 | ✅        |
| Unit tests                                                     | ✅        |
| Acceptance tests                                               | ✅        |
| Code format checks                                             | ✅        |
| Validate metadata for source-google-ads                        | ✅        |
| Connector version semver check                                 | ✅        |
| Connector version increment check                              | ✅        |
| QA checks                                                      | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/tolik0_source-google-ads_custom-query-fix/1695890472/d5e7f6b60efc7d31acf2556bff730735509073d1/source-google-ads/1.0.0/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:d5e7f6b60efc7d31acf2556bff730735509073d1)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-google-ads test
```

clnoll reviewed (approved):
LGTM once the breaking changes doc is approved!

pedroslopez commented:
/approve-and-merge reason="release + skip stuck gradle check"

octavia-approvington commented:
Enough! Let's do this <br/> ![goooooo!](https://media4.giphy.com/media/3oz8xHY5TPG9CAW0xi/giphy.gif?cid=ecf05e47z0yx82769b47apuo7ktdy80smghysn94klxp2k9v&rid=giphy.gif&ct=g)
